### ISSUE TITLE
tests: kernel: smp fatal exception observed on ESP32

### ISSUE THREAD

**Issue by spoorthik** [zephyrproject-rtos/zephyr#7787]
**Test location:** tests/kernel/smp

```
**Execution log:**
** FATAL EXCEPTION
 ** CPU 0 EXCCAUSE 29 PS 0x00060f20 PC 0x40082f27 VADDR 0x00000004
 **  A0 0x80082444  SP 0x3ffe3d40  A2 0x00000000  A3 0x00000001
 **  A4 0x00000001  A5 0x00000000  A6 0x00000000  A7 0x00000000
 **  A8 0x3ffb3500  A9 0x00000000 A10 0x3ffb20b4 A11 0x40082c2c
 ** A12 0x00000000 A13 0x00000000 A14 0x00000000 A15 0x3ffb20d8
 ** LBEG 0x4000c2e0 LEND 0x4000c2f6 LCOUNT 0x00000000
 ** SAR 0x00000005
Current thread ID = 0x3ffe7db0
Faulting instruction address = 0xdeaddead
Fatal fault in ISR! Spinning...

```
When I did git bisect, I got build error on 1acd8c2996c3a1ae0691247deff8c32519307f17, after proceeding by marking 1acd8c2996c3a1ae0691247deff8c32519307f17 as bad commit, it finally pointed at the same commit as first bad commit.

**Comment by andyross** [comment]
Yeah, I'm realizing that the sched patch broke esp32 in a few different ways, actually.  On it.

**Comment by MaureenHelm** [comment]
@andyross Any update?

**Comment by nashif** [comment]
resolved by #8040
